source/aggregation.txt

```diff
@@ -1,17 +1,265 @@
+- Sort the results
+
+You can use aggregation operations to perform the following actions:
+
+- Perform find operations
+- Rename fields
+- Calculate fields
+- Summarize data
+- Group values
+
+Limitations
+~~~~~~~~~~~
 
-  >>> from pymongoexplain import ExplainableCollection
-  >>> ExplainableCollection(collection).aggregate(pipeline)
-  {'ok': 1.0, 'queryPlanner': [...]}
+Keep the following limitations in mind when using aggregation operations:
 
-Or, use the the ``~pymongo.database.Database.command`` method method:
+- Returned documents must not violate the
+  :manual:`BSON document size limit </reference/limits/#mongodb-limit-BSON-Document-Size>`
+  of 16 megabytes.
+- Pipeline stages have a memory limit of 100 megabytes by default. You can exceed this
+  limit by using the ``allowDiskUse`` keyword argument of the
+  ``aggregate()`` method.
+
+.. important:: $graphLookup exception
+
+   The :manual:`$graphLookup
+   </reference/operator/aggregation/graphLookup/>` stage has a strict
+   memory limit of 100 megabytes and ignores the ``allowDiskUse`` parameter.
+
+Aggregation Example
+-------------------
+
+.. note::
+  
+  This example uses the ``sample_restaurants.restaurants`` collection
+  from the :atlas:`Atlas sample datasets </sample-data>`. To learn how to create a
+  free MongoDB Atlas cluster and load the sample datasets, see the :ref:`<pymongo-get-started>`.
+
+To perform an aggregation, pass a list of aggregation stages to the
+``collection.aggregate()`` method.
+
+The following code example produces a count of the number of bakeries in each borough
+of New York. To do so, it uses an aggregation pipeline with the following stages:
+
+- A :manual:`$match </reference/operator/aggregation/match/>` stage to filter for documents
+  whose ``cuisine`` field contains the value ``"Bakery"``.
+
+- A :manual:`$group </reference/operator/aggregation/group/>` stage to group the matching
+  documents by the ``borough`` field, accumulating a count of documents for each distinct
+  value of ``borough``.
 
 .. code-block:: python
+   :copyable: true
+
+   # Define an aggregation pipeline with a match stage and a group stage
+   pipeline = [
+       { $match: { cuisine: "Bakery" } },
+       { $group: { _id: "$borough", count: { $sum: 1 } } }
+   ]
+
+   # Execute the aggregation
+   aggCursor = collection.aggregate(pipeline)
+
+   # Print the aggregated results
+   for document in aggCursor:
+       print(document)
+
+The preceding code example produces output similar to the following:
+
+.. code-block:: javascript
+   
+   {'_id': 'Bronx', 'count': 71}
+   {'_id': 'Brooklyn', 'count': 173}
+   {'_id': 'Missing', 'count': 2}
+   {'_id': 'Manhattan', 'count': 221}
+   {'_id': 'Queens', 'count': 204}
+   {'_id': 'Staten Island', 'count': 20}
+
+Explain an Aggregation
+~~~~~~~~~~~~~~~~~~~~~~
+
+To view information about how MongoDB executes your operation, you can
+ask MongoDB to **explain** it. When MongoDB explains an operation, it returns
```

Comment:
S: This is nitpicky but I think "prompt" is more accurate than "ask" here

```suggestion
prompt MongoDB to **explain** it. When MongoDB explains an operation, it returns
```

Reply:
used "instruct"